### PR TITLE
fix: android layout bugs

### PR DESCRIPTION
**Pull request by Fyuryy**
# What I did
 - Fixed android related bugs such as the position of the lowbar whenever the keyboard is opened. Now the lowbar is well fixed throughout all the registration workflow. Moreover, when the keyboard is opened, it hides correctly the screen without pushing it and the lowbar is hidden.
 - Limited the date of birth to the date of the actual day
 - Passwords now do not appear in plain text but with <b>*</b>'s

# How I did it
Used useEffect and listeners to hide the keyboard and changed the zIndex of some components so they are hidden by the keyboard when it's opened.  

# How to verify it
By going through the registration flow like an user would, it's possible to check all the changes I made. 

# Pre-merge checklist
The changes I have introduced:
- [x] work correctly
- [x] do not break other functionalities
- [x] work correctly on Android
- [x] are fully tested

### CODE REVIEW

File: screens/Onboarding/styles.ts [hunk 1]
```diff
@@ -1,88 +1,71 @@
...
+    width: "100%",
   },
   buttonGoogle: {
-    alignItems: 'center',
     backgroundColor: white,
     borderRadius: BUTTON_RADIUS,
     borderWidth: 1,
-    display: 'flex',
-    flexDirection: 'row',
+    display: "flex",
```
Comment (Aidasvenc) [inline screens/Onboarding/styles.ts:38]: i am pretty sure that if you delete `display: "flex"`, the button is going to look the same way

File: screens/Onboarding/styles.ts [hunk 2]
```diff
@@ -1,88 +1,71 @@
...
-  footer: {
-    display: 'flex',
-    flexDirection: 'row',
-    position: 'absolute',
 
+  footer: {
+    flex: 1,
+    flexDirection: "row",
+    position: "absolute",
+    zIndex: -1,
```
Comment (Aidasvenc) [inline screens/Onboarding/styles.ts:66]: Why did you need a `zIndex: -1` here? Wasn't `bottom: insets.bottom` not enough for the footer to be anchored to the bottom?

File: screens/Registration/InformationScreen/InformationScreen.tsx [hunk 3]
```diff
@@ -30,8 +30,32 @@ const InformationScreen: React.FC = () => {
...
+    })
+    const hideSubscription = Keyboard.addListener("keyboardDidHide", () => {
+      setKeyboardVisible(false)
+    })
+
+    return () => {
+      showSubscription.remove()
+      hideSubscription.remove()
+    }
+  }, [])
```
Comment (AlbertoCentonze) [inline screens/Registration/InformationScreen/InformationScreen.tsx:47]: This seems to be recurring, why don't you create a custom hook called `useKeyboard` to access this more easily? I don't think you've ever made custom hook but it's really easy can help you out if needed!

File: screens/Registration/AuthenticationScreen/AuthenticationScreen.tsx [hunk 4]
```diff
@@ -116,24 +135,32 @@ const AuthenticationScreen: React.FC = () => {
...
+          ></InputField>
+        </View>
 
-        <LowBar nextScreen="HomeTabs" buttonText="Confirm" authenticate={submitForm} />
+        <View style={[styles.footer, { bottom: insets.bottom }]}>
+          {!keyboardVisible && <LowBar
+            nextScreen="HomeTabs"
+            buttonText="Confirm"
+            authenticate={submitForm}
+          />}
```
Comment (Aidasvenc) [inline screens/Registration/AuthenticationScreen/AuthenticationScreen.tsx:162]: Why do you want to hide the LowBar when keyboard is on?

**Review by AlbertoCentonze — CHANGES_REQUESTED** [review]
Hey, nice job!

Since I've seen a lot of `absolute` used in the CSS I would recommend testing the layout on different screen sizes since absolute can easily create footguns in this kind of scenario. I think you can safely do this by playing around with the window size of a react native web instance, so that you don't have to install a bunch of simulators.

File: screens/Registration/AuthenticationScreen/AuthenticationScreen.tsx (hunk 4 above)
Comment (Aidasvenc) [reply screens/Registration/AuthenticationScreen/AuthenticationScreen.tsx:162]: Sorry for a duplicate solution, but take a look at my code in #165 for how i anchored the LowBar to the bottom. Ideally, you should not wrap the LowBar in a View each time, but rather change the styling for the LowBar itself. Take a look at my solution to see more.

File: screens/Registration/AuthenticationScreen/styles.ts [hunk 5]
```diff
@@ -1,52 +1,61 @@
...
-    alignItems: 'center',
-    alignSelf: 'center',
+    alignItems: "center",
+    alignSelf: "center",
     backgroundColor: lightPeach,
     borderColor: peach,
     borderRadius: BUTTON_RADIUS,
     borderWidth: 3,
-    display: 'flex',
+    display: "flex",
```
Comment (Aidasvenc) [inline screens/Registration/AuthenticationScreen/styles.ts:13]: Same here

**Review by Aidasvenc — COMMENTED** [review]
Nice fixes! Inspect a couple of details regarding the styling and you will be good to go

File: screens/Onboarding/styles.ts (hunk 2 above)
Comment (Fyuryy) [reply screens/Onboarding/styles.ts:66]: The footer was anchored at the bottom without the zIndex: -1 indeed. The problem was when opening the keyboard, the footer came up. So I solved it this way. Do you think it would be better to do as with the other screens, with a useEffect?

File: screens/Registration/AuthenticationScreen/AuthenticationScreen.tsx (hunk 4 above)
Comment (Fyuryy) [reply screens/Registration/AuthenticationScreen/AuthenticationScreen.tsx:162]: The reason I want to hide it is because on IOS, the LowBar was totally fixed but on android the lowbar was fixed until the keyboard was opened and pushed it on top of it

File: screens/Onboarding/styles.ts (hunk 1 above)
Comment (Fyuryy) [reply screens/Onboarding/styles.ts:38]: Indeed

File: screens/Registration/AuthenticationScreen/styles.ts (hunk 5 above)
Comment (Fyuryy) [reply screens/Registration/AuthenticationScreen/styles.ts:13]: Indeed

File: screens/Registration/InformationScreen/InformationScreen.tsx (hunk 3 above)
Comment (Fyuryy) [reply screens/Registration/InformationScreen/InformationScreen.tsx]: Done!

**Review by AlbertoCentonze — APPROVED** [review]
